config/config_test.go

```diff
@@ -0,0 +1,284 @@
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package config
+
+import (
+	"fmt"
+	"os"
+	"testing"
+)
+
+// CreateFile creates yaml file for test purpose
+func CreateFile() {
+	fileData1 := `
+environment: minikube
+`
+	f, err := os.Create("./test-config.yaml")
+	if err != nil {
+		fmt.Println("Unable to create config file")
+	}
+	f.WriteString(fileData1)
+
+	// Create yaml file with bad intendation
+	fileData2 := `
+	environment: minikube
+	`
+	f, err = os.Create("./test-bad-config.yaml")
+	if err != nil {
+		fmt.Println("Unable to create bad config file")
+	}
+	f.WriteString(fileData2)
+
+}
+
+// DeleteFile deletes yaml file
+func DeleteFile() {
+	err := os.Remove("./test-config.yaml")
+	if err != nil {
+		fmt.Println("Unable to delete config file")
+	}
+
+	err = os.Remove("./test-bad-config.yaml")
+	if err != nil {
+		fmt.Println("Unable to delete bad config file")
+	}
+}
+
+func TestLoadConf(t *testing.T) {
+	CreateFile()
+	type args struct {
+		confFilePath string
+	}
+	tests := []struct {
+		name    string
+		args    args
+		wantErr bool
+	}{
+		{
+			name: "LoadConfFileNotPresent",
+			args: args{
+				confFilePath: "./file.yaml",
+			},
+			wantErr: true,
+		},
+		{
+			name: "LoadConfSuccess",
+			args: args{
+				confFilePath: "./test-config.yaml",
+			},
+			wantErr: false,
+		},
+		{
+			name: "LoadConfEmptyFileName",
+			args: args{
+				confFilePath: "",
+			},
+			wantErr: false,
+		},
+		{
+			name: "LoadConfBadYamlFile",
+			args: args{
+				confFilePath: "./test-bad-config.yaml",
+			},
+			wantErr: true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if err := LoadConf(tt.args.confFilePath); (err != nil) != tt.wantErr {
+				t.Errorf("LoadConf() error = %v, wantErr %v", err, tt.wantErr)
+			}
+		})
+	}
+	DeleteFile()
+}
+
+func Test_getConfValueByStringField(t *testing.T) {
+	type args struct {
+		conf  Configuration
+		field string
+	}
+	tests := []struct {
+		name string
+		args args
+		want string
+	}{
+		{
+			name: "KeyPresentInYaml",
+			args: args{
+				conf: Configuration{
+					Environment: "minikube",
+				},
+				field: "Environment",
+			},
+			want: "minikube",
+		},
+		{
+			name: "KeyNotPresentInYaml",
+			args: args{
+				conf: Configuration{
+					Environment: "minikube",
+				},
+				field: "environment",
+			},
+			want: "<invalid Value>",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if got := getConfValueByStringField(tt.args.conf, tt.args.field); got != tt.want {
+				fmt.Println(got)
```

Comment:
isn't it redundant? we are already printing what we `got` in next line.

Reply:
Yes! Removing it.

Reply:
cool